### ISSUE TITLE
Cuda code

### ISSUE THREAD

**Issue by haskarb** [nathanrooy/Particle-Swarm-Optimization-with-Python#1]
Thanks for awesome code. 
1. Can you write the CUDA code in python?
2. From where you picked up this algorithm.

**Comment by nathanrooy** [comment]
You could definitely use something like PyCUDA (https://developer.nvidia.com/pycuda) and code this fairly easily. But unless your optimizing with millions of particles (which is inefficient), I'm not sure why a GPU implementation would be needed? It would be much wiser to parallelize your cost function onto the GPU rather than the optimizer...

**Comment by haskarb** [comment]
Yes, I am using PyCuda to implement PSO. Actually, I wanted to use PSO in place of gradient descent in neural network. So, trying to modify your code to pyCUDA.

**Comment by nathanrooy** [comment]
Ahh, I did this same thing a few years ago! I ended up just transferring the network weights from the GPU back to the main computer memory. You do loose some time in the transfer but it wasn't the end of the world. How many weight values are you dealing with?

**Comment by haskarb** [comment]
Actually, right now I am focussing on PSO- CUDA version using Pycuda no where in the git has worked on it, also there  are not sufficient pyCUDA forums. In the later part I'll modify NN architecture.
